yaksh/file_utils.py

```diff
@@ -50,3 +50,30 @@ def extract_files(zip_file, path=None):
         zip_file.close()
         return zfiles, extract_path
 
+
+def is_csv(document):
+    try:
+        try:
+            content = document.read(1024).decode('utf-8')
+        except AttributeError:
+            document.seek(0)
+            content = document.read(1024)
+        sniffer = csv.Sniffer()
+        dialect = sniffer.sniff(content)
+        document.seek(0)
+    except (csv.Error, UnicodeDecodeError):
+        return False
+    return True
```

Comment:
Cant we just add `except AttributeError` as different exception instead of nesting. 
For Ex. `try then except UnicodeDecodeError except AttributeError`

Reply:
I had tried it as you can see in my commits, but then does not fulfill the requirement so kept nested.
Though can remove nested but will have two separate try..except.